django/db/models/sql/query.py

```diff
@@ -1021,6 +1021,17 @@ def change_aliases(self, change_map):
                 if alias == old_alias:
                     table_aliases[pos] = new_alias
                     break
+
+        # 3. Rename the external aliases of combined queries.
```

Comment:
I guess the comment here could be adjusted to mention that we're doing that both for direct and combined queries external aliases.